cmd/compare-namespace/compare-namespace.go

```diff
@@ -0,0 +1,218 @@
+	return prs
+}
+
+// decodeFile for kube secrets and return namespaces that the secret is attached to
+func decodeFile() ([]*hclwrite.Block, error) {
+	var blocks []*hclwrite.Block
+
+	data, err := os.ReadFile(mm.File)
+
+	if err != nil {
+		return nil, fmt.Errorf("error reading file %s", err)
+	}
+
+	f, diags := hclwrite.ParseConfig(data, mm.File, hcl.Pos{
+		Line:   0,
+		Column: 0,
+	})
+
+	if diags.HasErrors() {
+		return nil, fmt.Errorf("error getting TF resource: %s", diags)
+	}
+	blocks = f.Body().Blocks()
+
+	return blocks, nil
+}
+
+// resouceType will search for namespace in all resources in a Pull Request
+func resourceType(block *hclwrite.Block) (string, string) {
+	var resourceName string
+	var namespaceBlock string
+	var namespaceVar string
+
+	metadata := block.Body().Blocks()
+	for _, m := range metadata {
+		if m.Type() == "metadata" {
+			for key, attr := range m.Body().Attributes() {
+				if key == "name" {
+					expr := attr.Expr()
+					exprTokens := expr.BuildTokens(nil)
+					var valueTokens hclwrite.Tokens
+					valueTokens = append(valueTokens, exprTokens...)
+					resourceName = strings.TrimSpace(string(valueTokens.Bytes()))
+				}
+				if key == "namespace" {
+					expr := attr.Expr()
+					exprTokens := expr.BuildTokens(nil)
+					var valueTokens hclwrite.Tokens
+					valueTokens = append(valueTokens, exprTokens...)
+					namespaceBlock = strings.TrimSpace(string(valueTokens.Bytes()))
+					if strings.Contains(namespaceBlock, "var.") {
+						ns := strings.SplitAfter(namespaceBlock, ".")
+						varNamespace, err := varFileSearch(ns[1])
+						if err != nil {
+							log.Fatal(err)
+						}
+						namespaceVar = varNamespace
+					} else {
+						namespaceVar = namespaceBlock
+					}
+				}
+			}
+		}
+	}
+	return namespaceVar, resourceName
+}
+
+// varFileSearch will search for the namespace in the variables.tf file if the search returns a var.namespace
+func varFileSearch(ns string) (string, error) {
+	path := strings.SplitAfter(mm.File, "resources/")
+	data, err := os.ReadFile(path[0] + "variables.tf")
+	if err != nil {
+		return "", fmt.Errorf("error reading file %s", err)
+	}
+
+	v, diags := hclwrite.ParseConfig(data, path[0]+"variables.tf", hcl.Pos{
+		Line:   0,
+		Column: 0,
+	})
+
+	if diags.HasErrors() {
+		return "", fmt.Errorf("error getting TF resource: %s", diags)
+	}
+
+	var varNamespace string
+
+	blocks := v.Body().Blocks()
+	for _, block := range blocks {
+		if block.Type() == "variable" {
+			for _, label := range block.Labels() {
+				if label == ns {
+					for key, attr := range block.Body().Attributes() {
+						if key == "default" {
+							expr := attr.Expr()
+							exprTokens := expr.BuildTokens(nil)
+							var varTokens hclwrite.Tokens
+							varTokens = append(varTokens, exprTokens...)
+							varNamespace = strings.TrimSpace(string(varTokens.Bytes()))
+						}
+					}
+				}
+			}
+		}
+	}
+	return varNamespace, nil
+}
+
+func main() {
+	if *token == "" {
+		client = github.NewClient(nil)
```

Comment:
Is it worth logging here that we are init-ing a client without a token that could hit github rate limits?